packages/next/client/image.tsx

```diff
@@ -276,6 +292,40 @@ export default function Image({
   if (!configEnableBlurryPlaceholder) {
     placeholder = 'empty'
   }
+  const isStatic = typeof src === 'object'
+  let staticSrc = ''
+  if (isStatic) {
+    let staticImport = src as StaticImport
+    let staticImageData: StaticImageData
+    if ((staticImport as StaticRequire).default) {
+      staticImageData = (staticImport as StaticRequire).default
+    } else {
+      staticImageData = staticImport as StaticImageData
+    }
+    if (!staticImageData.src) {
+      throw new Error(
+        `An object should only be passed to the image component src parameter if it comes from a static image import. It must include src. Received ${JSON.stringify(
+          staticImageData
+        )}`
+      )
+    }
+    if (staticImageData.dataURI) {
+      blurDataURL = staticImageData.dataURI
+    }
+    staticSrc = staticImageData.src
+    if (!layout || layout !== 'fill') {
+      height = staticImageData.height
+      width = staticImageData.width
```

Comment:
Will these images look low quality on high DPI (retina) screens?

Usually the width/height are smaller than the source image and we render 1x & 2x for retina, but this code will use the source image's width/height so we can't upscale that to 2x 🤔

Reply:
As we discussed, we'll leave it as is for now, for fast prototyping reasons. Users can manually define a size if they want different behavior